tests/validation/tests/v3_api/resource/terraform/longliving/aws/v1apps/main.tf

```diff
@@ -0,0 +1,213 @@
+terraform {
+    required_providers {
+        rancher2 = {
+            source  = "rancher/rancher2"
+            version = "1.10.6"
+        }
+    }
+}
+
+provider "aws" {
+    region  = "us-east-2"
+    profile = "rancher-eng"
+}
+
+provider "rancher2" {
+    api_url    = var.cattle_test_url
+    token_key  = var.admin_token
+    insecure   = true
+}
+
+locals {
+    node_config = ["controlplane", "etcd", "worker", "worker", "worker"]
+}
+
+// create custom cluster in Rancher
+resource "rancher2_cluster" "custom-cluster" {
+    name = "custom-cluster"
+    description = "custom-cluster from terraform"
+    rke_config {
+        addon_job_timeout = 45
+        cloud_provider {
+            name = "aws"
+            aws_cloud_provider {
+            }
+        }
+        network {
+            plugin = "canal"
+        }
+        upgrade_strategy {
+            drain = false
+        }
+        services {
+            etcd {
+                backup_config {
+                    enabled = true
+                    retention = 3
+                    interval_hours = 6
+                    s3_backup_config {
+                        bucket_name = var.minio_bucket
+                        endpoint = var.minio_endpoint
+                        access_key = var.minio_access_key
+                        secret_key = var.minio_secret_key
```

Comment:
@git-ival and I noticed that you only need the endpoint and bucket_name for the s3 backup config. Trying to set the access key and secret was causing weird issues.